xamarin/realm-tutorial-dotnet/TaskPage.xaml.cs

```diff
@@ -0,0 +1,141 @@
+
+        public ObservableCollection<Task> MyTasks
+        {
+            get
+            {
+                return _tasks;
+            }
+        }
+
+        public TaskPage()
+        {
+            InitializeComponent();
+        }
+
+        protected override async void OnAppearing()
+        {
+            WaitingLayout.IsVisible = true;
+            try
+            {
+                taskRealm = await Realm.GetInstanceAsync();
+                SetUpTaskList();
+            }
+            catch (Exception ex)
+            {
+                await DisplayAlert("Error Fetching Tasks", ex.Message, "OK");
+            }
+            base.OnAppearing();
+        }
+
+        private void SetUpTaskList()
+        {
+            if (_tasks != null || _tasks.Count == 0)
+            {
+                WaitingLayout.IsVisible = true;
+                _tasks = new ObservableCollection<Task>(taskRealm.All<Task>().ToList());
+                listTasks.ItemsSource = _tasks;
+                WaitingLayout.IsVisible = false;
+            }
+        }
+
+        async void TextCell_Tapped(object sender, ItemTappedEventArgs e)
+        {
+            var task = e.Item as Task;
+            var editTaskPage = new EditTaskPage(taskRealm, task);
+            await Navigation.PushAsync(editTaskPage);
+        }
+
+        async void Edit_Button_Clicked(object sender, EventArgs e)
+        {
+            string result = await DisplayPromptAsync("New Task", "Enter the Task Name");
+
+            if (result == null)
+            {
+                return;
+            }
+
+            if (taskRealm == null)
+            {
+                taskRealm = await Realm.GetInstanceAsync();
+            }
+
+            var newTask = new Task()
+            {
+                Summary = result,
+                IsComplete = false
+            };
+
+            taskRealm.Write(() =>
+            {
+                taskRealm.Add(newTask);
+            });
+
+            MyTasks.Add(newTask);
+        }
+
+        void chkCompleted_PropertyChanged(object sender, PropertyChangedEventArgs e)
+        {
+            var isCompleteSwitch = (Switch)sender;
+            var changedTask = _tasks.FirstOrDefault(t => t.Id == isCompleteSwitch.AutomationId);
+            if (changedTask != null && e.PropertyName == "IsToggled")
+            {
+                taskRealm.Write(() =>
+                    {
+                        changedTask.IsComplete = isCompleteSwitch.IsToggled;
+                    });
+            }
+        }
+
+        async void Logout_Clicked(object sender, EventArgs e)
+        {
+            if (Navigation.NavigationStack.Count == 1)
+            {
+                var loginPage = new LoginPage();
+                NavigationPage.SetHasBackButton(loginPage, false);
+                await Navigation.PushAsync(loginPage);
+            }
+            else
+            {
+                await Navigation.PopToRootAsync();
+            }
```

Comment:
Does this method actually log out the user anywhere?

Reply:
@papafe is helping track down a bug I'm finding about that.

Reply:
fixed.